### ISSUE TITLE
Double click expand hidden engines

### ISSUE THREAD

**Issue by ZIEXED** [CanisLupus/swift-selection-search#59]
I want to add many more less used search engines, but to have a large list with many less used engines isn't an option

**Feature request**

- Double clicking empty space in the popup expand the popup list with more engines

in the settings page at "Search engines" a column named "Expand" before the column "Name"
and under it a check box beside every search engine to mark the engines as hidden in the popup
**in short** 
**Engines marked as hidden appears only after expanding the popup**

_I have other awesome suggestions :D should I open a new issue for every one or should I edit this issue and list them here?_

after testing many many extensions doing the same as SSS (including extensions at Chrome Store)
I was using another extension for a month until yesterday when I finally found SSS
in the end SSS is the best I have ever tested among all of them
I'm impressed with this masterpiece!

**Comment by CanisLupus** [comment]
Welcome, ZIEXED! :) Thanks for the suggestion and kind words. Woohoo "masterpiece" haha. You're the best! ;)

The suggestion sounds useful and I'll certainly consider it! Though I should say that I have no timeline right now, since I haven't had much time to work on SSS these last weeks.

To be clear, do you mean creating a new column of checkboxes or using the existing one? With the existing checkboxes, the hidden engines (i.e. those that would only appear when expanded) would simply be the ones that have their checkbox disabled. This would avoid adding a new column of checkboxes specifically, and simply let every engine appear if the popup has expanded. :)

As for other suggestions, if you don't mind creating a different issue for each one it would be better for keeping track! :)

Cheers!
Daniel

**Comment by ZIEXED** [comment]
I meant a new column of checkboxes (I thought someone may need to keep some engines as completely disabled), but if that would bring this feature to SSS a bit sooner... :D 

_I wish you all the happiness in the world ♥_

**Comment by CanisLupus** [comment]
Hehehe I see, although I would expect that people would simply delete the really unused ones (or back them up to a file, which is a possibility now), but maybe not. :p

Either way, I'll give it some thought. ;)

Cheers! <3

**Comment by ZIEXED** [comment]
Is there is a chance for this to get implemented, it has been 3 years and until today still I do really wish that I can do this.

**Comment by CanisLupus** [comment]
Hi @ZIEXED, sorry that there was no progress here in so long. I understand that feeling! I also haven't worked on SSS since about May, so no news for now,

There is a chance as long as the issue is here, :) I just have to prioritize based on perceived usefulness, bloat added to the settings page, and difficulty. JavaScript and the process of testing a Firefox WebExtension make SSS development 10x more painful than pretty much any other thing I code day to day in "not-web languages", and I have had other things going on since May. Shame on me for creating a Firefox addon hahah. :p

**Comment by CanisLupus** [comment]
Just now, thinking about this feature again, I realized that in SSS the webpages don't even get information (including icons) about disabled engines in order to save time and memory.

Either SSS would send everything based on an option (something like "Allow expanding hidden engines"), or I introduce async behaviour (an extra message between page and background script when expanding).

Also, this would probably have to be a button in the addon and not clicking empty space, which is hard to do and depends on your visual settings.

Then it raises the question of if icons like copy to clipboard and open link (SSS "special" icons) should also show up if disabled. Or if each engine should get an individual setting and the aforementioned button used to expand would only appear if you had one engine hidden... Decisions, decisions. :)

**Comment by CanisLupus** [comment]
By the way, I think I should look at a few smaller issues from here for an SSS update soon, but unfortunately this feature will _probably_ not be one of them, as I haven't found a comfortable way to implement it that doesn't overly complicate things. Don't lose hope yet, though! ;)

**Comment by ZIEXED** [comment]
Ok :), I understand it will take time and I hope it will eventually be implemented, just wanted to remind you of this issue as the number of open issues keep growing and it felt like it become forgotten.

**Comment by CanisLupus** [comment]
Yeah, reminding me that you have a use for the feature is a decent way to boost it. ;)
